pruning/architecture/pruning_loop.py

```diff
@@ -6,18 +6,23 @@
 from architecture.construct_dataset import get_dataloaders
 from architecture.construct_model import construct_model, register_models
 from architecture.construct_optimizer import construct_optimizer
+from architecture.pruning_methods.iterators import get_iterator
 import architecture.utility as utility
 from config.main_config import MainConfig
+import numpy as np
 import pandas as pd
 import torch
 from torch import nn
 import torch.nn.utils.prune as prune
 from wandb.sdk.wandb_run import Run
 
 logger = logging.getLogger(__name__)
+PRUNING_CLASSES = (nn.Linear, nn.Conv2d, nn.BatchNorm2d)
 
 
 def start_pruning_experiment(cfg: MainConfig, out_directory: Path) -> None:
+    assert cfg.pruning.iterator.end < 1, "The pruning iterator end value must be less than 1"
```

Comment:
I would move this to iterator or remove it, wouldn't store logic like that here.

Reply:
done